### PR TITLE
Image filters using 3-D convolution

### PR DESCRIPTION
**Pull request by Namyalg**
This script can be used to perform 3-D convolution on colored  images. 

Issue related to #93

### CODE REVIEW

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+r_kernel = np.array(list(map(int, input("Enter the kernel to be applied to the red stream").split()))).reshape(3,3)
+g_kernel = np.array(list(map(int, input("Enter the kernel to be applied to the green stream").split()))).reshape(3,3)
+
+#Image is read and resized
+
+img = cv2.imread('path to image')
+img = cv2.resize(img, (width , height))
+
+#The pixels are extracted from the image
+pixels = []
```
Comment (disha2sinha) [inline Image-Processing/Image_filters_using_convolution/3D_convolve.py:30]: Insert a new line before for loop

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                r.append(i[2])
+
+#Each of the streams are resized
+r = np.array(r).reshape(height, width)
+b = np.array(b).reshape(height, width)
+g = np.array(g).reshape(height, width)
+
+#function to apply convolution or multiply kernels
+def convolute(arr , height , width ,kernel):
+      convoluted_matrix = []
```
Comment (disha2sinha) [inline Image-Processing/Image_filters_using_convolution/3D_convolve.py:50]: Insert new line before for loop

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+for i in range(height):
+      for j in range(width):
+            combine[i ,j] = [int(r_convol[i][j]) , int(g_convol[i][j]) , int(b_convol[i][j])]
+
+
+#The 3 matrices or streams are combined, and stored
+img = Image.fromarray(combine)
+img.save('Funkyfilter.jpg')
+
+
```
Comment (disha2sinha) [inline Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Remove new lines at the end

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+width = width - 3
+height = height - 3
+
+combine = np.zeros([height, width, 3], dtype=np.uint8)
+
+for i in range(height):
+      for j in range(width):
+            combine[i ,j] = [int(r_convol[i][j]) , int(g_convol[i][j]) , int(b_convol[i][j])]
```
Comment (disha2sinha) [inline Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Remove space before comma and put space after comma

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+#function to apply convolution or multiply kernels
+def convolute(arr , height , width ,kernel):
+      convoluted_matrix = []
+      for i in range(1,height-2):
+          for j in range(1,width-2):
+            temp = arr[i:i+3 , j:j+3]
+            prod = np.multiply(temp,kernel)
+            convoluted_matrix.append(np.sum(prod))
+
+      convoluted_matrix = (np.array(convoluted_matrix).reshape(height-3,width-3))
```
Comment (disha2sinha) [inline Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Put space after comma

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+r = np.array(r).reshape(height, width)
+b = np.array(b).reshape(height, width)
+g = np.array(g).reshape(height, width)
+
+#function to apply convolution or multiply kernels
+def convolute(arr , height , width ,kernel):
+      convoluted_matrix = []
+      for i in range(1,height-2):
+          for j in range(1,width-2):
+            temp = arr[i:i+3 , j:j+3]
```
Comment (disha2sinha) [inline Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Remove space before comma and put space after comma

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            prod = np.multiply(temp,kernel)
```
Comment (disha2sinha) [inline Image-Processing/Image_filters_using_convolution/3D_convolve.py]: put space after comma

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+#Each of the streams are resized
+r = np.array(r).reshape(height, width)
+b = np.array(b).reshape(height, width)
+g = np.array(g).reshape(height, width)
+
+#function to apply convolution or multiply kernels
+def convolute(arr , height , width ,kernel):
+      convoluted_matrix = []
+      for i in range(1,height-2):
```
Comment (disha2sinha) [inline Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Put space after comma

**Review by disha2sinha — CHANGES_REQUESTED** [review]

**Comment by Namyalg** [comment]
I will make the changes right away.

**Review by disha2sinha — CHANGES_REQUESTED** [review]
Some of the changes requested earlier are yet to be made

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py (hunk 1 above)
Comment (disha2sinha) [reply Image-Processing/Image_filters_using_convolution/3D_convolve.py:30]: Make this change

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py (hunk 2 above)
Comment (disha2sinha) [reply Image-Processing/Image_filters_using_convolution/3D_convolve.py:50]: Make this change

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py (hunk 7 above)
Comment (disha2sinha) [reply Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Make this change

**Review by disha2sinha — CHANGES_REQUESTED** [review]

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py (hunk 5 above)
Comment (disha2sinha) [reply Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Make this change

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py (hunk 4 above)
Comment (disha2sinha) [reply Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Make this change

File: Image-Processing/Image_filters_using_convolution/3D_convolve.py (hunk 3 above)
Comment (disha2sinha) [reply Image-Processing/Image_filters_using_convolution/3D_convolve.py]: Make changes

**Comment by Namyalg** [comment]
Yes, I have made them

**Comment by Namyalg** [comment]
I would want to know if I can submit scripts that I have written using the PyGame module and raise an issue for the same

**Comment by powerexploit** [comment]
> I would want to know if I can submit scripts that I have written using the PyGame module and raise an issue for the same

Great work @Namyalg !
You can submit scripts written using pygame by raising new issue.

**Comment by Namyalg** [comment]
Okay, will do.

**Comment by Namyalg** [comment]
I would like to know if I have to make any changes to these scripts

**Review by disha2sinha — APPROVED** [review]

**Pull request merged**